### ISSUE TITLE
Incompatibility with "Remove Reloading Screen" in MC versions before 1.20.4

### ISSUE THREAD

**Issue by Jadan1213** [Keksuccino/FancyMenu#747]
**Describe the bug**
When using the mod "remove reloading screen" which reloads resources in the background without being stuck on a reloading screen, if you go to the main menu, Fancy menu crashes. If you wait for the resource reload to complete before going back to the main title screen, there is no crash.

**Game Log**
https://gist.github.com/Jadan1213/f9ff4cd6558f0d7a87c47d83d37138b1

**Basic Informations (please complete the following information):**
 - OS: Windows 10
 - FancyMenu Version: 3.1.2
 - Forge/Fabric Version : Forge 47.2.17
 - Minecraft Version: 1.20.1
 - Active Mods: a lot....

**Comment by Keksuccino** [comment]
Try to remove **Seamless Loading Screen**.

**Comment by Jadan1213** [comment]
> Try to remove **Seamless Loading Screen**.

seamless loading screen would have nothing to do with it, that just shows a screenshot of the last place you logged out while loading in, and fades into the world. (i disabled just to test, and it still crashed).

I'll save us some troubleshooting time. Made a barebones instance and the crash still happened.

latest.log: https://gist.github.com/Jadan1213/c85d1bbcc525f5e0b3987d8bf1baa1a0
Crash log: https://gist.github.com/Jadan1213/db455bcca2efe6df4876d5d8503cf886
Debug log: https://gist.github.com/Jadan1213/d8ce697ef161637933bf99de696f5088

**Comment by Jadan1213** [comment]
I just noticed your thread that you opened regarding incompatibility. I also use Not enough crashes, so it's not a game breaking thing when it happens. I just click back to title, or if i'm not loaded into a world, i just wait for the resources to finish reloading before clicking back to title so it doesn't crash. But hopefully my logs could be helpful.

**Comment by Keksuccino** [comment]
This instance is not "barebones", you had multiple mods installed.

```
[23Mar2024 05:36:50.275] [main/INFO] [Puzzles Lib/]: Loading 36 mods:
	- badpackets 0.4.3
	- balm 7.2.2
	- clienttweaks 11.1.0
	- cloth_config 11.1.118
	- craftingtweaks 18.2.3
	- dimensionteleport 2.0.1
	- easyshulkerboxes 8.0.2
	- embeddium 0.3.9+mc1.20.1
	- embeddiumplus 1.2.5
	- emi 1.1.3+1.20.1+forge
	- emi_loot 0.6.5+1.20.1+forge
	- emiffect 1.1.2+mc1.20.1
	- emitrades 1.2.1+mc1.20.1
	- entityculling 1.6.2
	- fancymenu 3.1.2
	- ferritecore 6.0.1
	- forge 47.2.17
	- immediatelyfast 1.2.10+1.20.4
	- jade 11.8.0
	- jei 15.3.0.4
	- konkrete 1.8.0
	- melody 1.0.2
	- memoryleakfix 1.1.5
	- minecraft 1.20.1
	- mixinextras 0.3.5
	- modernfix 5.14.0+mc1.20.1
	- mousetweaks 2.25
	- puzzlesaccessapi 8.0.9
	- puzzlesapi 8.1.6
	- puzzleslib 8.1.18
	- reeses_sodium_options 1.0.4
	- rrls 3.2.0-1.20.1-forge
	- rubidium 0.7.1
	- smoothboot 0.0.4
	- starlight 1.1.2+forge.1cda73c
	- textrues_embeddium_options 0.1.5+mc1.20.1
````

It that case the crash is probably caused by **Remove Reloading Screen**, because I tried to fix an incompatibility with it in the past but exactly this "fix" seems to crash the game now because something in RRLS changed probably.

**Comment by Keksuccino** [comment]
oh nvm, this is a duplicate of #729 ^^

**Comment by Jadan1213** [comment]
> The crash is probably caused by **Remove Reloading Screen**, because I tried to fix an incompatibility with it in the past but exactly this "fix" seems to crash the game now because something in RRLS changed probably.
> 
> ```
> [23Mar2024 05:36:50.275] [main/INFO] [Puzzles Lib/]: Loading 36 mods:
> 	- badpackets 0.4.3
> 	- balm 7.2.2
> 	- clienttweaks 11.1.0
> 	- cloth_config 11.1.118
> 	- craftingtweaks 18.2.3
> 	- dimensionteleport 2.0.1
> 	- easyshulkerboxes 8.0.2
> 	- embeddium 0.3.9+mc1.20.1
> 	- embeddiumplus 1.2.5
> 	- emi 1.1.3+1.20.1+forge
> 	- emi_loot 0.6.5+1.20.1+forge
> 	- emiffect 1.1.2+mc1.20.1
> 	- emitrades 1.2.1+mc1.20.1
> 	- entityculling 1.6.2
> 	- fancymenu 3.1.2
> 	- ferritecore 6.0.1
> 	- forge 47.2.17
> 	- immediatelyfast 1.2.10+1.20.4
> 	- jade 11.8.0
> 	- jei 15.3.0.4
> 	- konkrete 1.8.0
> 	- melody 1.0.2
> 	- memoryleakfix 1.1.5
> 	- minecraft 1.20.1
> 	- mixinextras 0.3.5
> 	- modernfix 5.14.0+mc1.20.1
> 	- mousetweaks 2.25
> 	- puzzlesaccessapi 8.0.9
> 	- puzzlesapi 8.1.6
> 	- puzzleslib 8.1.18
> 	- reeses_sodium_options 1.0.4
> 	- rrls 3.2.0-1.20.1-forge <----------------------------------------
> 	- rubidium 0.7.1
> 	- smoothboot 0.0.4
> 	- starlight 1.1.2+forge.1cda73c
> 	- textrues_embeddium_options 0.1.5+mc1.20.1
> ```

To respond to your comment you seem to have edited
![image](https://github.com/Keksuccino/FancyMenu/assets/68805162/23212465-ac30-4a97-842b-a13d1054ba59)

compared to my 500+ modpack that i run, yes, this is barebones.  It's a small selection of performance mods and client tweak mods that i use to test when other mods crash. this lets me test without having 500 mods loaded. It's easy to disable a few mods in a small pack to test an incompatability, it's not so easy with 500 of them.

**Comment by Jadan1213** [comment]
> Oh nvm, this is a duplicate of #729. Will keep this one open and close the other one instead, because this one has better logs, etc.

alright, if you need anyone to test any fixes, i'd be happy to try and help.

**Comment by Keksuccino** [comment]
## Update

The upcoming FancyMenu update will make "Remove Reloading Screen" fully incompatible with FancyMenu on MC versions **before 1.20.4**!

Both mods work fine together in MC 1.20.4+!

**Comment by Jadan1213** [comment]
I appreciate you backporting the fix!!

**Comment by Keksuccino** [comment]
Uhm that's not what I said, please read the message again :D

**Comment by Jadan1213** [comment]
Oh, well damn lol. I must have missed that when I read it.

**Comment by Keksuccino** [comment]
Fixed/Added in FancyMenu v3.1.3!
